controllers/raycluster_controller.go

```diff
@@ -0,0 +1,292 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/sha1"
+	"encoding/base64"
+
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	coreapply "k8s.io/client-go/applyconfigurations/core/v1"
+	v1 "k8s.io/client-go/applyconfigurations/meta/v1"
+	rbacapply "k8s.io/client-go/applyconfigurations/rbac/v1"
+	"k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	routev1 "github.com/openshift/api/route/v1"
+	routeapply "github.com/openshift/client-go/route/applyconfigurations/route/v1"
+	routev1client "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+)
+
+// RayClusterReconciler reconciles a RayCluster object
+type RayClusterReconciler struct {
+	client.Client
+	kubeClient  *kubernetes.Clientset
+	routeClient *routev1client.RouteV1Client
+	Scheme      *runtime.Scheme
+	CookieSalt  string
+}
+
+const (
+	requeueTime             = 10
+	controllerName          = "codeflare-raycluster-controller"
+	oauthAnnotation         = "codeflare.dev/oauth=true"
+	CodeflareOAuthFinalizer = "codeflare.dev/oauth-finalizer"
```

Comment:
May I suggest something like `openshift.ai/oauth-proxy`? In case we move the controller outside codeflare.